planet/src/main/webapp/js/itinerary.js

```diff
@@ -96,17 +181,48 @@ document.addEventListener('DOMContentLoaded', function() {
     let instances = M.FormSelect.init(elems, {});
 });
 
-// jQuery function that reorders the events
+// jQuery function that reorders the events on the front end
 $(function() { 
     $( "#events" ).sortable({
-    update: function(event, ui) { 
-        reorderEvents(ui); 
-    }       
+        update: function() { 
+            reorderEvents(); 
+        }       
     }); 
 }); 
-          
-function reorderEvents(ui) { 
+
+// Reorder in firebase
+function reorderEvents() { 
     $('.event').each(function (i) {
-        console.log('order is ' + i + ' for ' + this.innerHTML);
+        const userId = firebase.auth().currentUser.uid;
+        const eventRef = database.ref('users/' + userId +'/currentList/' + this.id);
+        eventRef.update({
+            order: i+1
+        });
     });
 } 
+
+// This is a class representing a time range, it mirrors the TimeRange class in the server
```

Comment:
nit: move this class to its own file.

Reply:
In order to import one class to another class, I need to declare my javascript files as modules instead of standard scripts in html. I tried it but my it won't load my script (if I call a function in the html file, it would say the function is not defined). I did some research and found that each module has their own scope, so the html file won't be able to use functions in the javascript module unless I call all the functions in js or I made every function global. I think both options require many changes that may take very long and was wondering if I can keep the class inside in this case.

Here are the resources I found:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules
https://stackoverflow.com/questions/53630310/use-functions-defined-in-es6-module-directly-in-html
![image](https://user-images.githubusercontent.com/46226072/86973693-ffb0ca80-c142-11ea-9e3c-8b23e8f072c2.png)

Reply:
I'm interested in exploring this further. There must be some way that we can set this up to allow us to organize our code.

However, I don't think we need to block this PR from proceeding.